Shared/Realm.Shared/Realm.cs

```diff
@@ -831,6 +831,16 @@ public RealmObject ObjectForPrimaryKey(string className, string id)
         /// <exception cref="System.ArgumentNullException">If you invoke this with a standalone object.</exception>
         public void Remove(RealmObject obj)
         {
+            if (obj == null)
+            {
+                throw new ArgumentNullException(nameof(obj));
+            }
+
+            if (!this.Equals(obj.Realm))
+            {
+                throw new RealmObjectManagedByAnotherRealmException("This object is managed by another realm. You should call Remove on the other realm.");
```

Comment:
It could also be standalone, right? So maybe just "This object is not managed by this realm"